erpc_c/transports/erpc_spi_slave_transport.cpp

```diff
@@ -149,41 +155,43 @@ erpc_status_t SpiSlaveTransport::underlyingSend(const uint8_t *data, uint32_t si
     slaveXfer.txData = (uint8_t *)data;
     slaveXfer.rxData = NULL;
     slaveXfer.dataSize = size;
+    {
 #else
     uint8_t *spiData = new (nothrow) uint8_t[size + ERPC_BOARD_SPI_SLAVE_READY_MARKER_LEN];
     if (spiData != NULL)
     {
         spiData[0] = ERPC_BOARD_SPI_SLAVE_READY_MARKER1;
         spiData[1] = ERPC_BOARD_SPI_SLAVE_READY_MARKER2;
         memcpy(&spiData[ERPC_BOARD_SPI_SLAVE_READY_MARKER_LEN], data, size);
-    }
-    else
-    {
-        return kErpcStatus_SendFailed;
-    }
-
-    slaveXfer.txData = spiData;
-    slaveXfer.rxData = NULL;
-    slaveXfer.dataSize = size + ERPC_BOARD_SPI_SLAVE_READY_MARKER_LEN;
+        slaveXfer.txData = spiData;
+        slaveXfer.rxData = NULL;
+        slaveXfer.dataSize = size + ERPC_BOARD_SPI_SLAVE_READY_MARKER_LEN;
 #endif
 
-    status = SPI_SlaveTransferNonBlocking(m_spiBaseAddr, &s_s_handle, &slaveXfer);
+        status = SPI_SlaveTransferNonBlocking(m_spiBaseAddr, &s_s_handle, &slaveXfer);
 
-    if (kStatus_Success == status)
-    {
+        if (kStatus_Success == status)
+        {
 #ifdef ERPC_BOARD_SPI_SLAVE_READY_USE_GPIO
-        SpiSlaveTransport_NotifyTransferGpioReady();
+            SpiSlaveTransport_NotifyTransferGpioReady();
 #endif
-        while (!s_isTransferCompleted)
-        {
-        }
+            while (!s_isTransferCompleted)
+            {
+            }
 #ifdef ERPC_BOARD_SPI_SLAVE_READY_USE_GPIO
-        SpiSlaveTransport_NotifyTransferGpioCompleted();
+            SpiSlaveTransport_NotifyTransferGpioCompleted();
 #endif
+        }
+#ifdef ERPC_BOARD_SPI_SLAVE_READY_USE_GPIO
+    }
+#else ERPC_BOARD_SPI_SLAVE_READY_USE_GPIO
```

Comment:
#else ERPC_BOARD_SPI_SLAVE_READY_USE_GPIO
will not work

Reply:
Do you mean that i need remove redundant word: ERPC_BOARD_SPI_SLAVE_READY_USE_GPIO? Will do

Reply:
to be honest, I am not clear about this part of code adjustment, what misra rule is being solved, anyway, both erpc_spi_slave_transport.cpp and erpc_dspi_slave_transport.cpp should be handled the same way (no such a change in erpc_dspi_slave_transport.cpp)

Reply:
Multiple returns in code and replaced #ifndef ERPC_BOARD_SPI_SLAVE_READY_USE_GPIO condition